packages/module/src/ListManager/ListManager.tsx

```diff
@@ -0,0 +1,170 @@
+  /** Internal identifier of a column by which table displayed columns are filtered. */
+  key: string;
+  /** The actual display name of the column possibly with a tooltip or icon. */
+  title: React.ReactNode;
+  /** If user changes checkboxes, the component will send back column array with this property altered. */
+  isSelected?: boolean;
+  /** Set to false if the column should be hidden initially */
+  isShownByDefault: boolean;
+  /** The checkbox will be disabled, this is applicable to columns which should not be toggleable by user */
+  isUntoggleable?: boolean;
+}
+
+export interface ListManagerProps {
+  /** Current column state */
+  columns: ListManagerItem[];
+  /** Custom OUIA ID */
+  ouiaId?: string | number;
+  /** Callback when a column is selected or deselected */
+  onSelect?: (column: ListManagerItem) => void;
+  /** Callback when all columns are selected or deselected */
+  onSelectAll?: (columns: ListManagerItem[]) => void;
+  /** Callback when the column order changes */
+  onOrderChange?: (columns: ListManagerItem[]) => void;
+  /** Callback to save the column state */
+  onSave?: (columns: ListManagerItem[]) => void;
+  /** Callback to close the modal */
+  onCancel?: () => void;
+}
+
+const ListManager: FunctionComponent<ListManagerProps> = (
+  { columns,
+    ouiaId = 'Column',
+    onSelect,
+    onSelectAll,
+    onOrderChange,
+    onSave,
+    onCancel }: ListManagerProps) => {
+
+  const [ currentColumns, setCurrentColumns ] = useState(
+    () => columns.map(column => ({ ...column, isSelected: column.isSelected ?? column.isShownByDefault, id: column.key }))
+  );
+
+
+  const handleChange = (columnKey: string) => {
+    const newColumns = [ ...currentColumns ];
+    const index = newColumns.findIndex(col => col.key === columnKey);
+    if (index === -1) {return;}
+    
+    const changedColumn = { ...newColumns[index] };
+    changedColumn.isSelected = !changedColumn.isSelected;
+    newColumns[index] = changedColumn;
+
+    setCurrentColumns(newColumns);
+    onSelect?.(changedColumn);
+  };
+
+  const onDrag = (_event: unknown, newOrder: any[]) => {
+    const newColumns = newOrder.map((item: any) => {
+      const found = currentColumns.find(c => c.key === item.id);
+      if (!found) {
+        throw new Error(`Column with key ${item.id} not found`);
+      }
+      return found;
+    });
+    setCurrentColumns(newColumns);
+    onOrderChange?.(newColumns);
+  };
+
+  const handleSave = () => {
+    onSave?.(currentColumns);
+  };
+
+  const handleBulkSelect = (value: BulkSelectValue) => {
+    const allSelected = value === 'all' || value === 'page';
+    handleSelectAll(allSelected);
+  };
+
+  const handleSelectAll = (select = true) => {
+    const newColumns = currentColumns.map(c => ({ ...c, isSelected: c.isUntoggleable ? c.isSelected : select }));
+    setCurrentColumns(newColumns);
+    onSelectAll?.(newColumns);
+  };
+
+  return (
+    <>
+      <div style={{ paddingBottom: '1rem' }}>
```

Comment:
can we update these styles to be `style={{ paddingBlockEnd: 'var(--pf-t--global--spacer--md)' }}`

double check that but it should be equivalent and more PF compliant.
PaddingBlockEnd makes it relative to the direction and layout of the international settings (for things like right to left languages).

the global spacer medium is the 1rem spacer but uses PF tokens in case people create themes in the future for PF with greater or lesser padding overall.